frontend/appflowy_flutter/lib/mobile/presentation/database/card/card_property_edit/type_option_widget_builder/number.dart

```diff
@@ -0,0 +1,158 @@
+              children: [
+                PropertyTitle(LocaleKeys.grid_field_numberFormat.tr()),
+                const Spacer(),
+                const HSpace(4),
+                Text(
+                  state.typeOption.format.title(),
+                  style: Theme.of(context).textTheme.titleMedium,
+                ),
+                Icon(
+                  Icons.arrow_forward_ios,
+                  color: Theme.of(context).hintColor,
+                ),
+              ],
+            ),
+            onTap: () => showFlowyMobileBottomSheet(
+              context,
+              isScrollControlled: true,
+              title: LocaleKeys.grid_field_numberFormat.tr(),
+              builder: (bottomsheetContext) => GestureDetector(
+                child: NumberFormatList(
+                  onSelected: (format) {
+                    context
+                        .read<NumberTypeOptionBloc>()
+                        .add(NumberTypeOptionEvent.didSelectFormat(format));
+                    bottomsheetContext.pop();
+                  },
+                  selectedFormat: state.typeOption.format,
+                ),
+              ),
+            ),
+          );
+        },
+      ),
+    );
+  }
+}
+
+typedef SelectNumberFormatCallback = Function(NumberFormatPB format);
+
+class NumberFormatList extends StatelessWidget {
+  final SelectNumberFormatCallback onSelected;
+  final NumberFormatPB selectedFormat;
+  const NumberFormatList({
+    required this.selectedFormat,
+    required this.onSelected,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => NumberFormatBloc(),
+      child: Column(
+        children: [
+          const _FilterTextField(),
+          const VSpace(16),
+          SizedBox(
+            height: 300,
+            child: BlocBuilder<NumberFormatBloc, NumberFormatState>(
+              builder: (context, state) {
+                final cells = state.formats.map((format) {
+                  return RadioListTile<NumberFormatPB>(
+                    controlAffinity: ListTileControlAffinity.trailing,
+                    visualDensity: VisualDensity.compact,
+                    value: format,
+                    groupValue: selectedFormat,
+                    onChanged: (format) {
+                      onSelected(format!);
+                    },
+                    title: Text(format.title()),
+                  );
+                }).toList();
+
+                final list = ListView.builder(
+                  itemCount: cells.length,
+                  itemBuilder: (BuildContext context, int index) {
+                    return cells[index];
+                  },
+                );
+                return list;
+              },
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _FilterTextField extends StatelessWidget {
+  const _FilterTextField({Key? key}) : super(key: key);
```

Comment:
Super.key constructor
```suggestion
  const _FilterTextField({super.key});

```

Reply:
I deleted it.
The same reason for the private widget, it cannot pass flutter analysis if we add it.